tests/puppeteer/test/linkchecker.js

```diff
@@ -0,0 +1,160 @@
+require('events').EventEmitter.defaultMaxListeners = Infinity;
+const puppeteer = require('puppeteer');
+const fs = require('fs');
+
+const reportPath = './reports/';
+// maybe add tracing option : https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-tracing
+const URL_FO = process.env.URL_FO || 'http://localhost:8080/';
+const URL_BO = process.env.URL_BO || 'http://localhost:8080/admin-dev/';
+const EMAIL = process.env.LOGIN || 'demo@prestashop.com';
+const PASSWD = process.env.PASSWD || 'prestashop_demo';
+const HEADLESS = process.env.HEADLESS || true;
+let responses = [];
+const numberRequests = {BO: 0, FO: 0};
+const selectorBO = 'nav.nav-bar.d-none.d-md-block ul li a.link[href]';
+const selectorFO = 'a[href]';
+let office = null;
+let curHref = null;
+const object = {
+  stats: {
+    start: new Date().getTime(),
+    end: null,
+  },
+  BO: {
+    totalLinks: 0,
+    totalRequests: 0,
+    passed: [],
+    failed: [],
+  },
+  FO: {
+    totalLinks: 0,
+    totalRequests: 0,
+    passed: [],
+    failed: [],
+  },
+};
+
+const getAllUrls = async (page, selector) => {
+  const listUrls = await page.$$eval(selector, as => as.map(a => a.href));
+  // remove perfect duplicates
+  await listUrls.filter((elem, pos) => listUrls.indexOf(elem) === pos);
+  return listUrls;
+};
+
+const interceptRequestAndResponse = async (page) => {
+  await page.setRequestInterception(true);
+  await page.on('request', (request) => {
+    request.continue();
+  });
+  await page.on('response', (response) => {
+    const request = response.request();
+    const url = request.url();
+    const status = response.status().toString();
+    numberRequests[office] += 1;
+    if (status.startsWith('4') || status.startsWith('5')) {
+      console.log(` !! Failed response : ${url} (${status})`);
```

Comment:
shouldn't be `console.warn()` or `console.error()`? Some tools are able to colorize the stack trace according to the level of the console output

Reply:
done, thanks